libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3642,6 +3642,14 @@ bool TypeChecker::visit(Identifier const& _identifier)
 				_identifier.location(),
 				"\"suicide\" has been deprecated in favour of \"selfdestruct\"."
 			);
+		else if (_identifier.name() == "selfdestruct" && fType->kind() == FunctionType::Kind::Selfdestruct)
+			m_errorReporter.warning(
+				5159_error,
+				_identifier.location(),
+				"\"selfdestruct\" has been deprecated. "
+				"The underlying opcode will eventually be removed from the EVM, "
```

Comment:
I don't think it is correct to say it will be removed. Most likely semantics of it will change, bu the instruction remains.

Reply:
Ah, good point. Looking at EIP-6049 now, it never really says that it's going to be removed. Just changed.